github/GithubObject.py

```diff
@@ -367,10 +408,27 @@ def _completeIfNeeded(self) -> None:
     def __complete(self) -> None:
         if self._url.value is None:
             raise IncompletableObject(400, message="Returned object contains no URL")
-        headers, data = self._requester.requestJsonAndCheck("GET", self._url.value)
+        headers, data = self._requester.requestJsonAndCheck("GET", self._url.value, headers=self.__completeHeaders)
         self._storeAndUseAttributes(headers, data)
         self.__completed = True
 
+    def do_complete_unless_lazy(self, lazy: Opt[bool]) -> Self:
```

Comment:
This could use it's own dedicated pydoc. Also why is `lazy` optional, but not given a default value of `None`